### PR TITLE
dbft: add message pool

### PR DESCRIPTION
**Pull request by roman-khimov**
Based on NeoGo, no extensibles, just pure dBFT data.

### CODE REVIEW

File: eth/protocols/dbft/payload.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+var invalidSig = errors.New("invalid signature")
+
+func (m Message) Hash() common.Hash {
+	var h common.Hash
+
+	m.Witness = nil // m is a copy and witness is not hashed.
+	hw := sha3.NewLegacyKeccak256()
+	rlp.Encode(hw, m)
```
Comment (AnnaShaleva) [inline eth/protocols/dbft/payload.go:38]: Don't we want to implement `rlp.Encoder` interface for proper payload encoding?

File: eth/protocols/dbft/payload.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+	Sender common.Address
+	// Data is custom payload data.
+	Data []byte
+	// Witness is payload signature.
+	Witness []byte
+}
+
+var invalidSig = errors.New("invalid signature")
+
+func (m Message) Hash() common.Hash {
```
Comment (AnnaShaleva) [inline eth/protocols/dbft/payload.go:33]: Do we pay attention to the coding style? Exported method doesn't have comment, but it's OK for initial implementation.

**Review by AnnaShaleva — COMMENTED** [review]
LGTM otherwise.

File: eth/protocols/dbft/payload.go (hunk 1 above)
Comment (roman-khimov) [reply eth/protocols/dbft/payload.go:38]: I'm not sure yet we need it, it's supposed to be used for types that require custom encoding rules and we have a simple structure here.

File: eth/protocols/dbft/payload.go (hunk 2 above)
Comment (roman-khimov) [reply eth/protocols/dbft/payload.go:33]: Fixed!

**Review by AnnaShaleva — APPROVED** [review]

**Pull request merged**